### PR TITLE
Added site sampling to 1D raveller

### PR DESCRIPTION
**Pull request by LiamBindle**
Added the ability to create a 1D track for sampling ground sites. For example

```console
$ python -m gcpy.raveller_1D create_site_track --csv /home/liam/Downloads/SPARTAN_sites.csv --samples_per_day 48 --separator "\t" -o spartan_sites.nc
```

where `SPARTAN_sites.csv` is a file like this (tab separator, specified by `--separator` argument and passed to pandas)
```
23.728	90.398	Dhaka
-6.888	107.610	Bandung
21.048	105.800	Hanoi
14.635	121.078	Manila
37.564	126.935	Seoul
40.004	116.326	Beijing
-25.75661	28.280	Pretoria
-34.555	-58.506	Buenos Aires
1.298	103.780	Singapore
44.638	-63.594	Halifax
43.790	-79.470	Toronto
26.513	80.232	Kanpur
8.484	4.675	Ilorin
31.907	34.811	Rehovot
45.380	-71.931	Sherbrooke
49.682	-112.869	Lethbridge
33.688	-84.290	Atlanta
49.941	-119.400	Kelowna
24.442	54.617	Abu Dhabi
37.132	-86.148	Mammoth Cave
40.053	-88.372	Bondville
4.637	-74.081	Palmira
```

### CODE REVIEW

File: gcpy/raveller_1D.py [hunk 1]
```diff
@@ -181,6 +223,29 @@ def has_converged(absdiff, curr_min, n=10):
                               required=True,
                               help='output filename')
 
+    create_site_track = subparsers.add_parser(
+            'create_site_track', help='Generate a track file for ground sites')
+    create_site_track.add_argument('--csv',
+                              type=str,
```
Comment (WilliamDowns) [inline gcpy/raveller_1D.py:229]: Can you align these arguments with the ones from the other add_argument() calls?

File: gcpy/raveller_1D.py [hunk 2]
```diff
@@ -181,6 +223,29 @@ def has_converged(absdiff, curr_min, n=10):
...
+    create_site_track.add_argument('--ts',
+                                   type=float,
+                                   default=1/3600,
+                                   help='time separator between site samples (should be small number greater than 0)')
+    create_site_track.add_argument('--samples_per_day',
+                                   type=int,
+                                   required=True,
+                                   help='number of samples to take per day')
+    create_site_track.add_argument('-o',
+                              type=str,
```
Comment (WilliamDowns) [inline gcpy/raveller_1D.py:245]: Also here

File: gcpy/raveller_1D.py [hunk 3]
```diff
@@ -181,6 +223,29 @@ def has_converged(absdiff, curr_min, n=10):
                               required=True,
                               help='output filename')
 
+    create_site_track = subparsers.add_parser(
+            'create_site_track', help='Generate a track file for ground sites')
+    create_site_track.add_argument('--csv',
+                              type=str,
+                              required=True,
```
Comment (WilliamDowns) [inline gcpy/raveller_1D.py:230]: Can you align these arguments with the ones from the other add_argument() calls?

**Review by WilliamDowns — CHANGES_REQUESTED** [review]
Can you change the target to `dev/1.0.3` instead of `main`?

**Comment by WilliamDowns** [comment]
Oops, review comments out of order.

**Comment by msulprizio** [comment]
@LiamBindle What is the status of this update? Should we reopen this PR and merge into `dev`?

**Pull request closed without merging**